### PR TITLE
Remove tempdir, add files via config-file by string and add tests

### PR DESCRIPTION
**Pull request by rikvdh**
Also performed some cleaning in some files.

Running the tests as root (e.g. in docker) the tests of tempdir failed, when looking at the code, it seems quite odd to have this in a Debian packaging tool.

### CODE REVIEW

File: config_test.go [hunk 1]
```diff
@@ -21,6 +22,12 @@ maintainer_email: deb@pkg.com
 homepage: https://github.com/xor-gate/debpkg
 section: devel
 priority: standard
+depends: lsb-release
+recommends: nano
+suggests: curl
+conflicts: pico
+provides: editor
+replaces: vim
```
Comment (xor-gate) [inline config_test.go]: mwhahaha 😈

File: config_test.go [hunk 2]
```diff
@@ -36,10 +43,19 @@ files:
   - file: debpkg_test.go
   - file: README.md
     dest: {{.DATAROOTDIR}}/foobar/README.md
+  - dest: /bin/hello
+    content: >
+      #!/bin/bash
+      echo "hello"
```
Comment (xor-gate) [inline config_test.go]: Looks good

File: config_test.go [hunk 3]
```diff
@@ -36,10 +43,19 @@ files:
...
 directories:
   - ./internal
 emptydirs:
   - /var/cache/foobar
+control_extra:
+  postrm: Makefile
+  prerm: Makefile
+  postinst: Makefile
+  preinst: Makefile
```
Comment (xor-gate) [inline config_test.go]: Intuitive

File: config_test.go [hunk 4]
```diff
@@ -68,6 +84,52 @@ emptydirs:
 	assert.Nil(t, testWrite(t, deb))
 }
 
+func TestExampleConfig2(t *testing.T) {
```
Comment (xor-gate) [inline config_test.go]: Maybe name it TestExampleConfigWithControlExtraContent? Config2 is not saying very much.

File: config_test.go [hunk 5]
```diff
@@ -107,3 +169,27 @@ func TestNonExistingConfig(t *testing.T) {
...
 }
+
+func TestInvalidYAML(t *testing.T) {
+	deb := New()
+	defer deb.Close()
+
+	const configFile = `name: debpkg
+	foo: bar
+	`
+	filepath, err := test.WriteTempFile("debpkg2.yml", configFile)
```
Comment (xor-gate) [inline config_test.go]: use t.Name(), we are using go 1.8 or higher for testing

File: control.go [hunk 6]
```diff
@@ -193,14 +208,19 @@ func (deb *DebPkg) SetBuiltUsing(info string) {
 
 // AddControlExtraString is the same as AddControlExtra except it uses a string input
```
Comment (xor-gate) [inline control.go]: Document it replaces dos line endings with unix for compat

File: deb.go [hunk 7]
```diff
@@ -20,19 +21,25 @@ type DebPkg struct {
 	err          error
 }
 
-// New creates new debian package
-func New() *DebPkg {
+// New creates new debian package, optionally provide an tempdir to write
+//  intermediate files, otherwise os.TempDir is used.
+func New(tempDir ...string) *DebPkg {
```
Comment (xor-gate) [inline deb.go]: Good compat, much better then the tempdir builtin which was to complicated. When tempdir doesn't exist. It is not automaticly created, maybe this is a nice feature to have (e.g build pipeline with custom tempdir which doesn't exist).

Comment (rikvdh) [reply deb.go]: I don't like the automatic directory-creation. It shouldn't be a job of debpkg, perhaps I can document it.

**Pull request merged**